Xamarin.PropertyEditing/OrderedDictionary.cs

```diff
@@ -303,6 +333,8 @@ public void Insert (int index, TKey key, TValue value)
 		{
 			this.keyOrder.Insert (index, key);
 			this.dict.Add (key, value);
```

Comment:
Not new to this change, but I think this can also throw if you try to insert a key that already exists.

Reply:
It should, duplicate keys are invalid.

Reply:
Yes, but in a way that potentially corrupts state if that exception is handled up the call stack. What I'm saying is that it may be better to explicitly handle the case up-front and throw before you attempt to change state. Otherwise, I think the `Insert` may succeed and the `Add` fail, resulting in inconsistent state.

Reply:
Ah yeah fair point, I can swap their order.

Reply:
In fact, swapping them just produces the same concern in a different way (now key is added if index is out of range.) We have to check both prior.